FreeSurfer/LongitudinalFreeSurferPipeline.sh

```diff
@@ -0,0 +1,418 @@
+
+  # break FreeSurfer version into components
+  # primary, secondary, and tertiary
+  # version X.Y.Z ==> X primary, Y secondary, Z tertiary
+  freesurfer_version_array=(${freesurfer_version//./ })
+
+  freesurfer_primary_version="${freesurfer_version_array[0]}"
+  freesurfer_primary_version=${freesurfer_primary_version//[!0-9]/}
+
+  freesurfer_secondary_version="${freesurfer_version_array[1]}"
+  freesurfer_secondary_version=${freesurfer_secondary_version//[!0-9]/}
+
+  freesurfer_tertiary_version="${freesurfer_version_array[2]}"
+  freesurfer_tertiary_version=${freesurfer_tertiary_version//[!0-9]/}
+
+  if [[ $(( ${freesurfer_primary_version} )) -lt 6 ]]; then
+    # e.g. 4.y.z, 5.y.z
+    log_Err_Abort "FreeSurfer version 6.0.0 or greater is required. (Use FreeSurferPipeline-v5.3.0-HCP.sh if you want to continue using FreeSurfer 5.3)"
+  fi
+}
+
+# Configure the use of FreeSurfer v6 custom tools
+configure_custom_tools
+
+# Show tool versions
+show_tool_versions
+
+# Validate version of FreeSurfer in use
+validate_freesurfer_version
+
+# ----------------------------------------------------------------------
+log_Msg "Starting main functionality"
+# ----------------------------------------------------------------------
+
+
+# ----------------------------------------------------------------------
+# Log values retrieved from positional parameters
+# ----------------------------------------------------------------------
+log_Msg "StudyFolder: ${StudyFolder}"
+log_Msg "SubjectID: ${SubjectID}"
+log_Msg "Sessions: ${Sessions}"
+log_Msg "TemplateID: ${TemplateID}"
+log_Msg "recon_all_seed: ${recon_all_seed}"
+log_Msg "extra_reconall_args_base: ${extra_reconall_args_base[*]+"${extra_reconall_args_base[*]}"}"
+log_Msg "extra_reconall_args_long: ${extra_reconall_args_long[*]+"${extra_reconall_args_long[*]}"}"
+
+# ----------------------------------------------------------------------
+log_Msg "Preparing the folder structure"
+# ----------------------------------------------------------------------
+Sessions=`echo ${Sessions} | sed 's/@/ /g'`
+log_Msg "After delimiter substitution, Sessions: ${Sessions}"
+LongDIR="${StudyFolder}/${SubjectID}.long.${TemplateID}/T1w"
+mkdir -p "${LongDIR}"
+for Session in ${Sessions} ; do
+  Source="${StudyFolder}/${Session}/T1w/${Session}"
+  Target="${LongDIR}/${Session}"
+  log_Msg "Creating a link: ${Source} => ${Target}"
+  ln -sf ${Source} ${Target}
+done
+
+
+if (( generate_timepoints_only != 1 )); then 
+
+	# ----------------------------------------------------------------------
+	log_Msg "Creating the base template: ${TemplateID}"
+	# ----------------------------------------------------------------------
+	# backup template dir if it exists
+	if [ -d "${LongDIR}/${TemplateID}" ]; then
+	  TimeStamp=`date +%Y-%m-%d_%H.%M.%S.%6N`
+	  log_Msg "Base template dir: ${LongDIR}/${TemplateID} already exists, backing up to ${LongDIR}/${TemplateID}.${TimeStamp}"
```

Comment:
In general, throughout the pipelines, we don't worry about preserving existing copies of things.  That is the responsibility of users outside of the pipelines themselves.  So, do we want to start that now?  If we do, I'd prefer that we use the modification date on the existing `TemplateID` directory as the suffix to add, not the current date/time.

Reply:
I should have looked ahead.  I see that Tim makes the same comment below.

Reply:
Matt and I think the pipelines shouldn't be internally making copies of old results.

Reply:
Should they just delete old results or overwrite without notice to user?

Reply:
Overwrite without notice, as is the default with most commands.  If there is a substantial problem that could arise from mismatching some old and some new files, then it might make sense to delete.  I don't know if there are considerations here that need similar sanity checking to the hand-edit possibility in cross-sectional freesurfer.

Reply:
Ok, overwriting without notice